include/solvers/heat_transfer_assemblers.h

```diff
@@ -0,0 +1,213 @@
+
+#include <solvers/copy_data.h>
+#include <solvers/heat_transfer_scratch_data.h>
+#include <solvers/multiphysics_interface.h>
+
+/**
+ * @brief A pure virtual class that serves as an interface for all
+ * of the assemblers for heat transfer
+ *
+ * @tparam dim An integer that denotes the number of spatial dimensions
+ *
+ * @ingroup assemblers
+ */
+template <int dim>
+class HeatTransferAssemblerBase
+{
+public:
+  /**
+   * @brief assemble_matrix Interface for the call to matrix assembly
+   * @param scratch_data Scratch data containing the heat transfer information.
+   * It is important to note that the scratch data has to have been re-inited
+   * before calling for matrix assembly.
+   * @param copy_data Destination where the local_rhs and loc
+   */
+
+  virtual void
+  assemble_matrix(HeatTransferScratchData<dim> &scratch_data,
+                  StabilizedMethodsCopyData &   copy_data) = 0;
+
+  /**
+   * @brief assemble_matrix Interface for the call to rhs
+   * @param scratch_data Scratch data containing the heat transfer information.
+   * It is important to note that the scratch data has to have been re-inited
+   * before calling for matrix assembly.
+   * @param copy_data Destination where the local_rhs and loc
+   */
+
+  virtual void
+  assemble_rhs(HeatTransferScratchData<dim> &scratch_data,
+               StabilizedMethodsCopyData &   copy_data) = 0;
+};
+
+/**
+ * @brief Class that assembles the core of the heat transfer equation.
+ * This class assembles the weak form of:
+ * $$ - k * \nabla^2 T + \rho * cp * \mathbf{u} * \nabla T - f - \tau :
+ * \nabla \mathbf{u} =0 $$
+ *
+ * @tparam dim An integer that denotes the number of spatial dimensions
+ *
+ * @ingroup assemblers
+ */
+
+
+template <int dim>
+class HeatTransferAssemblerCore : public HeatTransferAssemblerBase<dim>
+{
+public:
+  HeatTransferAssemblerCore(
+    std::shared_ptr<SimulationControl> simulation_control,
+    Parameters::PhysicalProperties     physical_properties,
+    const SimulationParameters<dim> &  p_simulation_parameters)
+    : simulation_control(simulation_control)
+    , physical_properties(physical_properties)
+    , simulation_parameters(p_simulation_parameters)
+  {}
+
+  /**
+   * @brief assemble_matrix Assembles the matrix
+   * @param scratch_data (see base class)
+   * @param copy_data (see base class)
+   */
+  virtual void
+  assemble_matrix(HeatTransferScratchData<dim> &scratch_data,
+                  StabilizedMethodsCopyData &   copy_data) override;
+
+
+  /**
+   * @brief assemble_rhs Assembles the rhs
+   * @param scratch_data (see base class)
+   * @param copy_data (see base class)
+   */
+  virtual void
+  assemble_rhs(HeatTransferScratchData<dim> &scratch_data,
+               StabilizedMethodsCopyData &   copy_data) override;
+
+
+  std::shared_ptr<SimulationControl> simulation_control;
+  Parameters::PhysicalProperties     physical_properties;
+  const SimulationParameters<dim> &  simulation_parameters;
```

Comment:
What are simulation parameters and why are they needed? If you have simulation control and physical properties?